plugins/MumblePlugin_v_1_0_x.h

```diff
@@ -0,0 +1,408 @@
+// Copyright 2019-2020 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+/// This header file specifies the Mumble plugin interface
+
+#ifndef EXTERNAL_MUMBLE_PLUGIN_H_
+#define EXTERNAL_MUMBLE_PLUGIN_H_
+
+#include "PluginComponents_v_1_0_x.h"
+#include "MumbleAPI_v_1_0_x.h"
+#include <stdint.h>
+#include <stddef.h>
+#include <stdbool.h>
+
+#if defined(__GNUC__) && !defined(__MINGW32__) // GCC on Unix-like systems
+	#define PLUGIN_EXPORT __attribute__((visibility("default")))
+#elif defined(_MSC_VER)
+	#define PLUGIN_EXPORT __declspec(dllexport)
+#elif defined(__MINGW32__)
+	#define PLUGIN_EXPORT __attribute__((dllexport))
+#else
+	#error No PLUGIN_EXPORT definition available
+#endif
+
+// API version
+#define MUMBLE_PLUGIN_API_MAJOR_PREPR 1
+#define MUMBLE_PLUGIN_API_MINOR_PREPR 0
+#define MUMBLE_PLUGIN_API_PATCH_PREPR 0
+
+const int32_t MUMBLE_PLUGIN_API_MAJOR = MUMBLE_PLUGIN_API_MAJOR_PREPR;
+const int32_t MUMBLE_PLUGIN_API_MINOR = MUMBLE_PLUGIN_API_MINOR_PREPR;
+const int32_t MUMBLE_PLUGIN_API_PATCH = MUMBLE_PLUGIN_API_PATCH_PREPR;
+const mumble_version_t MUMBLE_PLUGIN_API_VERSION = { MUMBLE_PLUGIN_API_MAJOR, MUMBLE_PLUGIN_API_MINOR, MUMBLE_PLUGIN_API_PATCH };
+
+// Create macro for casting the pointer to the API object to the proper struct.
+// Note that this must only be used if the API uses MUMBLE_PLUGIN_API_VERSION of the API.
+#define MUMBLE_CONCAT_HELPER(a, b) a ## _ ## b
+#define MUMBLE_CONCAT(a, b) MUMBLE_CONCAT_HELPER(a, b)
+#define MUMBLE_API_STRUCT MUMBLE_CONCAT(MumbleAPI_v, MUMBLE_CONCAT(MUMBLE_PLUGIN_API_MAJOR_PREPR, MUMBLE_CONCAT(MUMBLE_PLUGIN_API_MINOR_PREPR, x)))
+#define MUMBLE_API_CAST(ptrName) ( *((struct MUMBLE_API_STRUCT *) ptrName) )
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+	//////////////////////////////////////////////////////////////////////////////////
+	////////////////////////////// MANDATORY FUNCTIONS ///////////////////////////////
+	//////////////////////////////////////////////////////////////////////////////////
+	
+	/// Gets called right after loading the plugin in order to let the plugin initialize.
+	///
+	/// Registers the ID of this plugin.
+	/// @param id The ID for this plugin. This is the ID Mumble will reference this plugin with
+	/// 	and by which this plugin can identify itself when communicating with Mumble.
+	/// @returns The status of the initialization. If everything went fine, return STATUS_OK
+	PLUGIN_EXPORT mumble_error_t PLUGIN_CALLING_CONVENTION mumble_init(uint32_t id);
+	
+	/// Gets called when unloading the plugin in order to allow it to clean up after itself.
+	PLUGIN_EXPORT void PLUGIN_CALLING_CONVENTION mumble_shutdown();
```

Comment:
For all I know it should be, yes. At least that's what is intended.

I was thinking that as long as it doesn't state that it is unsafe to do so, it is rather clear that it is allowed :thinking: 
I guess explicitly noting it doesn' hurt though

Reply:
Is it meant for things like closing files, closing db connections and the like?
Or are there any common things we need to do here?
( i do nothing right now, because i guess global variables will be decommissioned automatically once mumbles ram is handed back to the os)

Reply:
Yeah global variables have a managed lifetime. This function is meant to be used for cleaning up allocated resources. For instance if you allocate some stuff using `new` when the plugin starts, you have to call `delete` once you are done with it. And you should definitely be done with it once the plugin shuts down.